search-app/src/cmr/search/services/acls/collections_cache.clj

```diff
@@ -116,67 +116,34 @@
-    (if (and (= :return-errors throw-errors) (empty? collection-map))
-      (errors/internal-error! (coll-msg/collections-not-in-cache key-type))
-      (time-strs->clj-times collection-map)))))
-
-;; TODO remove after transition
-(defn get-collection
-  "Gets a single collection from the cache by concept id. Handles refreshing the cache if it is not found in it.
-  Also allows provider-id and entry-title to be used."
-  ([context concept-id]
-   (let [by-concept-id (get-collections-map context :by-concept-id :no-errors)]
-     (when-not (and by-concept-id (by-concept-id concept-id))
-       (info (coll-msg/collection-not-found concept-id))
-       (refresh-cache context))
-     (get (get-collections-map context :by-concept-id ) concept-id)))
+(defn get-collection-for-gran-acls
+  ([context coll-concept-id]
+   "Gets a single collection from the cache by concept id. If collection is not found, will add it to the cache (if it exists in elastic) and will return the found collection."
+   (let [coll-by-concept-id-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-concept-id-cache-key)
+         collection (hash-cache/get-value coll-by-concept-id-cache
+                                          coll-for-gran-acl-caches/coll-by-concept-id-cache-key
+                                          coll-concept-id)]
+     (if (or (nil? collection) (empty? collection))
+       (do
+         (info (str "Collection with concept-id " coll-concept-id " not found in cache. Will update cache and try to find."))
+         (time-strs->clj-times (coll-for-gran-acl-caches/set-caches context coll-concept-id)))
+       (time-strs->clj-times collection))))
   ([context provider-id entry-title]
-   (let [by-provider-id-entry-title (get-collections-map context :by-provider-id-entry-title )]
-     (get by-provider-id-entry-title [provider-id entry-title]))))
+   "Gets a single collection from the cache by provider-id and entry-title. If collection is not found, will add it to the cache (if it exists in elastic) and will return found collection."
+   (let [coll-by-provider-id-and-entry-title-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key)
+         collection (hash-cache/get-value coll-by-provider-id-and-entry-title-cache
+                                          coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key
+                                          (str provider-id entry-title))]
+     (if (or (nil? collection) (empty? collection))
+       (do
+         (info (str "Collection with provider-id " provider-id " and entry-title " entry-title " not found in cache. Will update cache and try to find."))
+         (time-strs->clj-times (coll-for-gran-acl-caches/set-caches context provider-id entry-title)))
+       (time-strs->clj-times collection)))))
 
 ;; added to common lib will remove in future
 (defjob RefreshCollectionsCacheForGranuleAclsJob
```

Comment:
Has this been added to common lib?  does this need to be removed?

Reply:
Next PR, I will remove all of these other funcs along with the entire in-memory coll cache